bot/cogs/eggs/reaction.py

```diff
@@ -0,0 +1,27 @@
+from discord.ext.commands import Cog
+from datetime import datetime, timedelta
+
+
+class Reaction(Cog, name="Eggs"):
+    def __init__(self, client):
+        self.bot = client
+        self.THRESHOLD = timedelta(seconds=1.5)
+        self.mapping = {}
+
+    @Cog.listener("on_reaction_add")
+    async def catch(self, reaction, user):
+        self.mapping.setdefault(user, {})[reaction.message] = datetime.now()
+
+    @Cog.listener("on_reaction_remove")
+    async def release(self, reaction, user):
```

Comment:
Since the only time you remove items from the dictionary is when a reaction is removed, there is a high possibility for a memory leak. I suggest implementing an automatic cleanup mechanism that invalidates items, such as a timer that removes items older than a certain timespan.